compositor_api/src/types/register_input.rs

```diff
@@ -122,8 +124,23 @@ pub enum InputRtpAudioOptions {
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
-#[serde(tag = "decoder", rename_all = "snake_case", deny_unknown_fields)]
-pub enum InputRtpVideoOptions {
-    #[serde(rename = "ffmpeg_h264")]
-    FfmepgH264,
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct InputRtpVideoOptions {
+    pub decoder: VideoDecoder,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub enum VideoDecoder {
+    /// Use the software decoder based on ffmpeg.
+    FfmpegH264,
+
+    /// Use hardware decoder based on Vulkan Video.
+    ///
+    /// This should be faster and more scalable than teh ffmpeg decoder, if the hardware and OS
+    /// support it.
+    ///
+    /// This requires hardware that supports Vulkan Video. Another requirement is this program has
+    /// to be compiled with the `vk-video` feature enabled.
```

Comment:
This might be confusing for someone who just wants to use precompiled binaries. Maybe we should mention what features are supported in the precompiled binaries. I would also add here mention like:
```rust
    /// This requires hardware that supports Vulkan Video. Another requirement is this program has
    /// to be compiled with the `vk-video` feature enabled (enabled by default on Linux x86, disabled by default on other platforms).
```